diff -Nur a/mixer.c b/mixer.c
--- a/mixer.c
+++ b/mixer.c
@@ -420,6 +420,10 @@
 		}
 	}

+	if (format == AUDIO_S16 || format == AUDIO_S16MSB || format == AUDIO_S16SYS)
+		format = AUDIO_S16LSB;
+	Mix_Init(MIX_INIT_OGG|MIX_INIT_MOD|MIX_INIT_MP3);
+
 	/* Set the desired format and frequency */
 	desired.freq = frequency;
 	desired.format = format;
@@ -476,6 +480,9 @@
 
 	audio_opened = 1;
 	SDL_PauseAudio(0);
+
+	Mix_Init(MIX_INIT_OGG|MIX_INIT_MOD|MIX_INIT_MP3);
+
 	return(0);
 }
